tests/test_vision_tagger_telemetry.py

```
@@ -0,0 +1,171 @@
+        assert payload["error"] == "fetch_timeout"
+        assert payload["image_size_bytes"] is None
+        assert payload["prompt_tokens"] is None
+
+    def test_fetch_rejected(self, caplog):
+        """`_fetch_image` raises ValueError for SSRF / oversize / non-image."""
+        caplog.set_level(logging.WARNING, logger=LOGGER_NAME)
+        with patch.object(
+            vision_tagger, "_fetch_image",
+            side_effect=ValueError("image_url is not allowed"),
+        ):
+            with pytest.raises(ValueError):
```

Ruff (0.15.12)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)